ui/util/url.js

```diff
@@ -0,0 +1,68 @@
+const LBRY_INC_DOMAINS = ['lbry.io', 'lbry.com', 'lbry.tv', 'lbry.tech', 'lbry.fund', 'spee.ch'];
+
+exports.formatLbryUrlForWeb = uri => {
+  return uri.replace('lbry://', '/').replace(/#/g, ':');
+};
+
+exports.formatFileSystemPath = path => {
+  if (!path) {
+    return;
+  }
+
+  let webUrl = path.replace(/\\/g, '/');
+
+  if (webUrl[0] !== '/') {
+    webUrl = `/${webUrl}`;
+  }
+
+  return encodeURI(`file://${webUrl}`).replace(/[?#]/g, encodeURIComponent);
+};
+
+exports.isLBRYDomain = uri => {
+  if (!uri) {
+    return;
+  }
+
+  const myURL = new URL(uri);
+  const hostname = myURL.hostname;
+
+  for (let domain of LBRY_INC_DOMAINS) {
+    if (hostname.endsWith(domain)) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+/*
+  Function that handles page redirects
+  ex: lbry://?rewards
+  ex: open.lbry.com/?rewards
+*/
+exports.formatCustomUrl = path => {
+  // Determine if we need to add a leading "/$/" for app pages
+  const APP_PAGE_REGEX = /(\?)([a-z]*)(.*)/;
+  const appPageMatches = APP_PAGE_REGEX.exec(path);
+
+  if (appPageMatches && appPageMatches.length) {
+    // Definitely an app page (or it's formatted like one)
+    const [, , page, queryString] = appPageMatches;
+    let actualUrl = '/$/' + page;
+
+    if (queryString) {
+      if (queryString.startsWith('?')) {
```

Comment:
do we still end up in this function with URLs containing two question marks? is this for malformed URLs / just-in-case, or is this specifically needed?

Reply:
Do you have an example url structure in mind I can test?

Reply:
I'm possibly confused, but wouldn't this only execute if `formatInAppUrl` received a `path` that contained two `?`?